### DISCUSSION TITLE
CMS: descrizione dei giochi e associazioni tra episodio e gioco

### DISCUSSION THREAD

**Discussion by pdonadeo** [pdonadeo/e-uscito-joypad#13] [General]
Sto lavorando alle nuove tabelle di dati che ci mancano ovvero i giochi e la associazione tra episodi e giochi.

**Non c'è ancora niente online.**

Questa è una piccola preview.

### Il gioco
Per ora l'unica cosa che ho inserito nella descrizione del gioco è... il titolo. Così:

![image](https://user-images.githubusercontent.com/242049/167948675-6003cf13-8fc7-4380-9592-2067305368f3.png)

Ok il titolo c'è... cosa volete mettere? Una URL alla "home page" del gioco? L'editore? Gli sviluppatori? Ad occhio io eviterei di aggiungere troppa roba perché ricordiamo che lo scopo è fare una ricerca _per titolo di IP_ quindi immagino che se uno cerca "Death Stranding" sappia già che l'autore è Kojima. Io lascerei forse solo il titolo e magari una cover sempre 256x256px. Se avete altre idee però ditelo.

### Associazione tra episodio e IP

Ovviamente è possibile associare i giochi agli episodi, anche più volte. L'associazione _per ora_ contiene come unica informazione l'istante in cui si è parlato del gioco. Nella scheda CMS dell'episodio ci sono tutti i giochi e nella scheda del gioco ci sono (automaticamente) tutti gli episodi in cui se ne è parlato. Forse qualche immagine è meglio.

Scheda dell'episodio:

![image](https://user-images.githubusercontent.com/242049/167950576-7b7f433b-8274-408b-ab71-fd04f24039ac.png)

Scheda del gioco:

![image](https://user-images.githubusercontent.com/242049/167950774-ce0c9da8-d36d-4e5a-9e2d-83e5629ab633.png)

Notate che è Elden ring è presente più volte associato ad un certo episodio, con minutaggi differenti.

Cosa possiamo aggiungere alla citazione? Chi ne ha parlato certamente... e poi? Fate sapere.

**Comment by MarcoJul** [comment]
Direi, a livello di informazione, di usare come informazione solo il titolo, almeno come primo step. Come dicevamo, mi immagino l'utente che appunto usa il campo di ricerca per vedere in quali episodi si é parlato del tale gioco, senza troppe altre informazioni.

Potrebbe essere uno step più avanzato quello di mettere, insieme alle puntate filtrate dalla ricerca per tale titolo, una scheda più o meno densa di informazioni (come url del gioco metterei una cosa tipo metacritic, o un qualsiasi database neutro nel caso).

Per quanto riguarda il tag, si usa un discrimine di "quantità"? Nel senso, inseriamo in database TUTTE le volte che viene citato il gioco o solo quelle in cui effettivamente é entrato in argomento? (magari a qualcuno non interessa che Death Stranding sia stato nominato come battuta in... beh tutte le puntate😅)

**Reply by pdonadeo** [reply]
Per la quantità di citazioni direi che è a discrezione e buon senso di chi compilerà il DB. Ovvio che se mettiamo tutte le volte che Bordone prende per il culo Zampa per Final Fantasy non ne usciamo vivi 😂

**Reply by bionicco** [reply]
Per la relazione direi che almeno un tag sulla tipologia di citazione è necessario (recensione/consiglio/citazione/classifica/altro per esempio) perchè magari mi interessa solo la recensione e non altro. Inoltre potrebbe essere carino un array con gli autori che ne parlano
(magari vado a sentire solo uno dei consigli di Bordone su hades, ma quello di Zampa invece lo voglio sentire)

Per il gioco il titolo è l'indispensabile. L'immagine non sarebbe male se vogliamo fare anche un'interfaccia che listi i titoli (a me piacerebbe).
Tutto il resto non so se sia davvero utile.
Però esistono delle API per collegarsi ad archivi di videogiochi e potrebbe essere la soluzione più semplice. Sarebbe sufficiente inserire l'ID del gioco e poi tutte le info che vogliamo mostrare le prendiamo da lì (e possiamo anche cambiare quali mostrare senza dover modificare i dati). 
Indago un po' la cosa:
https://rawg.io/apidocs
https://www.igdb.com/api

**Reply by pdonadeo** [reply]
Non conoscevo queste API, le guardo stasera: vediamo se tornano utili.

**Reply by bionicco** [reply]
RAWG mi sembra perfetto.
Ho richiesto una free api key e funziona bene!
Basta fare una ricerca per slug (che chiunque può ottenere facendo una ricerca sul loro sito e prelevandolo dall'url) e si hanno tantissime informazioni (piattaforme, uscita, metacritic, descrizione, sito, e molto altro).
Ovviamente dovrà essere citata la fonte dei dati sul sito, ma mi sembra un "prezzo" accettabile

PS: E' incredibile quante API esistono!

**Comment by pdonadeo** [comment]
@bionicco ho dato un'occhiata a RAWG e non sembra male ma oggi sono troppo stanco, se ne parla nel fine settimana.

**Comment by pdonadeo** [comment]
@bionicco sto implementando l'API di RAWG dentro al CMS. Il database sembra molto completo ma l'API fa veramente schifo. Comunque ormai la sto facendo, usiamo questa.

**Comment by agofa** [comment]
@pdonadeo sarebbe comodo avere i titoli in ordine alfabetico nella drop down da cui si sceglie il gioco da associare all'episodio
(nulla di critico ovviamente)

**Reply by pdonadeo** [reply]
Fatto, vedi #15 

Tra poco va on line.

Ad ogni modo suggerisco di non usare quella tendina, ma il tasto `+`

![image](https://user-images.githubusercontent.com/242049/168441640-c5644ca2-e8b7-44a0-8205-a67cb593a769.png)

**Reply by pdonadeo** [reply]
@agofa il drop-down ordinato per titolo è in line in questo momento.

**Reply by agofa** [reply]
E la peppa!
(grazie)

(e comunque il + col cavolo che l'avevo visto)
